packages/portal/src/Portal.tsx

```diff
@@ -19,8 +19,28 @@ function createPortalContainer(className?: string): HTMLElement {
 }
 
 function Portal(props: PortalProps) {
-  const [container] = useState(
-    props.container ?? createPortalContainer(props.className),
+  const [container, setContainer] = useState(props.container);
```

Comment:
I'm wondering if you could do:

```jsx
const [container] = useState(
    props.container ?? () => createPortalContainer(props.className));
```

As it should only run once and useState will call the anonymous function.

Reply:
See see the comment down below. The TL;DR is that it will run more than once in dev
https://github.com/facebook/react/issues/15074#issuecomment-471197572